### PR TITLE
v5.0.0.8 Candidate

### PR DESCRIPTION
**Pull request by MatthewVita**
(below added by bradymiller)
demo for this PR is at: https://www.open-emr.org/wiki/index.php/Development_Demo#Delta_-_Up_For_Grabs_Demo

### CODE REVIEW

**Comment by MatthewVita** [comment]
Some crazy formatting... but this is ready for my testing :)

**Comment by MatthewVita** [comment]
Testing approach...

```
$ git checkout v5_0_0_8_candidate
$ wget https://raw.githubusercontent.com/openemr/openemr/master/docker-compose.yml
$ cd contrib/util
$ mkdir docker
$ cd docker
$ wget https://raw.githubusercontent.com/openemr/openemr/master/contrib/util/docker/Dockerfile
$ wget https://github.com/openemr/openemr/raw/master/contrib/util/docker/docker-assets.zip
$ cd ../..
$ docker-compose up
```

**Comment by bradymiller** [comment]
hi @MatthewVita ,
Aren't you missing the libraries?
-brady

**Comment by bradymiller** [comment]
(also, you can ignore travis fails, since it does not work with 5.0.0 codebase)

**Comment by MatthewVita** [comment]
@bradymiller Ah ha! I thought I was missing a commit. Looks like it's https://github.com/openemr/openemr/commit/215c131cfcf749a5456ecaf4c493cedb647272c3

**Comment by bradymiller** [comment]
Set up and Up For Grabs demo for this PR at:
https://www.open-emr.org/wiki/index.php/Development_Demo#Delta_-_Up_For_Grabs_Demo

File: version.php [hunk 1]
```diff
@@ -11,7 +11,7 @@
 // A real patch identifier. This is incremented when we release a patch for a
 // production release. Note the above $v_patch variable is a misnomer and actually
 // stores release version information.
-$v_realpatch = '7';
+v_realpatch = '8';
```
Comment (bradymiller) [inline version.php]: @MatthewVita 
caught on up for grabs demo (unable to install secondary to parsing error here (the $ was removed).

Comment (MatthewVita) [reply version.php]: Ha, that would do it!

**Comment by bradymiller** [comment]
Just reset the demo (rather than waiting until tomorrow night), which can be tested now.

**Comment by MatthewVita** [comment]
Documents are blowing up:

```
[Thu Feb 22 04:38:56.663082 2018] [:error] [pid 77] [client 172.22.0.1:36156] PHP Fatal error:  Smarty error: [in /var/www/localhost/htdocs/openemr/templates/documents/general_list.html line 67]: syntax error: unrecognized tag: $(&quot;img[id^='icon_objTreeMenu_']&quot;).tooltip({\n        items: $(&quot;img[src*='file3.png']&quot;),\n        content: '{/literal (Smarty_Compiler.class.php, line 446) in /var/www/localhost/htdocs/openemr/vendor/smarty/smarty/libs/Smarty.class.php on line 1094, referer: http://localhost:8080/interface/patient_file/summary/demographics.php?set_pid=1&is_new=1
```

Good old syntax error!

**Comment by MatthewVita** [comment]
Okay, I fixed this by wrapping `{literal}` around the JS. Hope that's an okay practice

**Comment by MatthewVita** [comment]
Okay, I just pushed my changes. The _only_ thing I had to kill was the .tooltip. We don't have that as a bower dependency and I couldn't figure out what package it was that holds that. Was not obvious to me.

Would appreciate if @sjpadgett or @kofiav could do a real quick test on this branch. If you want to spin a docker is the fastest: https://github.com/openemr/openemr/pull/1429#issuecomment-366595835

**Comment by sjpadgett** [comment]
Thank goodness I have my 8-track of the Doobie Bro's to keep me calm or i'd be up on browser dev sites giving what for about you can't secure browser by restricting access for the dadnabit user(refer: security issues with downloads)! Trust me, for every attempt they make to secure browser, I bet it can be hacked, so knock it off... I swear poor dev's have to still put up with if this browser or that browser then this browser want this and that browser wants something else obscure finally please allow me to access this! Sorry, it needed to be said somewhere:) Although, for now, I'll reserve my wrath for the Florida state insurance commission:). 
Seems to be an intermittent issue (80/20 fail) with FF (between windows and linux) and file api detection. Fine on Chrome/Edge!

**Comment by MatthewVita** [comment]
@sjpadgett The latest commit to this branch has the fixes for Document php type and the js/template literal issue.

With respect to FF issues, what's version of FF are you on?

Also, it sounds like I should bring in the  jquery-ui widget, but I'm not sure it will work with the old version of jQuery that is on the rel-500 branch. Will look into that

**Comment by MatthewVita** [comment]
Okay, going to bring back in these tooltip lines on a fresh docker:

screen shot 2018-02-22 at 11 27 09 pm

Booooooo:

screen shot 2018-02-22 at 11 29 01 pm

This is bringing in jquery ui 11.1.4, which may be too old. I'll look into it

**Comment by MatthewVita** [comment]
Hmm, it does say "version added: 1.9" on the tooltip documentation.... hmmm.....

**Comment by sjpadgett** [comment]
Hi Matt, Have appt this a.m but will pull down branch and look later today. I still have concern with mime detection and file handling anyway.

**Comment by sjpadgett** [comment]
Haven't forgotten this just been busy.

**Comment by sjpadgett** [comment]
Am looking this over today. Had heck of a time getting to v5.0.0(7) with add in for viewer(8). @bradymiller @MatthewVita Do we still want do this?

**Comment by bradymiller** [comment]
@sjpadgett ,
Since 5.0.1 release is going out in 2 weeks, dont think is gonna be worth the effort. thoughts @MatthewVita ?

**Comment by sjpadgett** [comment]
It's already done. Fixed tooltip and pop out and normal display. I just checked out v5.0.0 to submit. So .....

**Comment by bradymiller** [comment]
If it's done, then bring it on :)

**Comment by MatthewVita** [comment]
@sjpadgett Wow! That's awesome. I can't find your PR here though: https://github.com/openemr/openemr/pulls/sjpadgett ?

**Comment by sjpadgett** [comment]
@MatthewVita Can I just push the changes to your branch. Having trouble with  my 5.0 branch?

**Comment by sjpadgett** [comment]
@MatthewVita 
Just pushed changes here. You were close but jquery and dlgopen were stepping on each other and jquery-ui 1-11-4 doesn't have the tooltip widget installed so I brought in version 1-12-1 from v5.0.1 branch.
Could use a test though to ensure I got everything.

**Comment by sjpadgett** [comment]
Okay, I installed this branch anew on windows and everything works as we/I expect. @bradymiller if, as the kind soul you are, will reset docker for this, I will test.

**Comment by MatthewVita** [comment]
@sjpadgett Cool, thanks for pushing. Here's what I see on my end:

dcm

**Comment by bradymiller** [comment]
@sjpadgett and @MatthewVita ,
I just reset the up for grabs demo to use the most recent code here :)

**Comment by sjpadgett** [comment]
Duh! v5.0.0 doesn't have jquery 3-1-1. Still mission focused on updating jquery regardless. :) Glad we dockered (new word) it. One more time to verify tool tip and popup view @bradymiller

**Comment by sjpadgett** [comment]
@MatthewVita Think should look into a routine to read dicom header because from what little I know about dicom from radiology the files are ship for their viewer with what patient the file belongs embedded in dicom. This makes user import into openemr tough. Maybe Pac's does this but i'm concerned how useful this is now. I know clinic I try this out on can't use because they don't know whom file belongs.

**Comment by bradymiller** [comment]
Sounds like we are a go on this :)
I'll bring this in and see if I can get some time to build a patch.

**Pull request merged**